Doc/library/struct.rst

```diff
@@ -400,6 +425,98 @@ alignment does not enforce any alignment.
       Packing and unpacking of XDR data.
 
 
+.. _applications:
+
+Applications
+------------
+
+Two main applications for the ``struct`` module exist, data
+interchange between Python and C within the application or with
+another application compiled using the same compiler (native formats),
+and data interchange between applications using agreed upon data
+layout (machine-independent formats).  Generally speaking, the format
+strings constructed for these two domains are distinct.
+
+
+Native Formats
+^^^^^^^^^^^^^^
+
+When constructing format strings which mimic native layouts, the
+compiler and machine architecture determine byte ordering and padding.
+In such cases, the ``@`` format character should be used to specify
+native byte ordering and data sizes.  Generally speaking, the ``l``
+format character should be used to specify the compiler's preferred
+long integer size.  This will vary across architectures.  For example,
+on a typical 64-bit architecture ``l`` specifies an 8-byte long
+integer.  On 32-bit architectures it most often specifies a 4-byte
+long integer.  Similarly, internal pad bytes are normally inserted
+automatically.  It's possible that a zero-repeat format code will be
+necessary at the end of a format string to round up to the correct
+byte boundary for proper alignment of consective chunks of data.
+
+Consider these two simple examples (on a 64-bit, little-endian
+machine)::
+
+    >>> len(struct.pack('@lhl', 1, 2, 3))
+    24
+    >>> len(struct.pack('@llh', 1, 2, 3))
+    18
+
+Data is not padded to an 8-byte boundary without the use of extra
+padding.  A zero-repeat format code solves that problem::
+
+    >>> len(struct.pack('@llh0l', 1, 2, 3))
+    24
+
+The ``x`` format code can be used to specify the repeat, but for
+native formats it's best to use a zero-repeat format like ``0l``.
+
+By default, native byte ordering and alignment is used, but it's best
+to be explicit and use the ``@`` format character.
+
+
+Machine-Independent Formats
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When exchanging data with other applications, byte ordering and
+alignment are specified precisely, and there is a good chance they
+won't correspond exactly to that of the local machine architecture.
+For example, network byte order is big-endian, while most commodity
```

Comment:
change "most commodity" to "many popular" as that is more accurate.

Reply:
Got it.